File: TP2/TP2Q07/StarWarsPesquisa.java
/*
Nome: Larissa Domingues Gomes
Turno: Manha
Numero de Matricula: 650525
Professor: Felipe Cunha 
Questao:Classe Personagens Star Wars Java Pesquisa Sequncial
*/

import java.util.Date;
import java.io.FileWriter;
import java.io.PrintWriter;

public class StarWarsPesquisa{
   public static void main(String[] args)throws Exception
   {
      long inicio = new Date().getTime();
      ListaPersonagem p = new ListaPersonagem(90); 
      FileWriter arquivo = new FileWriter("matrícula_sequencial.txt");
      PrintWriter gravar = new PrintWriter(arquivo);
      String arq = MyIO.readLine();

      //Repeticao para fazer o set de dados do array de personagens ate arq = fim
      while(!arq.equals("FIM")) {
         p.inserirFim(arq);
         arq = MyIO.readLine();
      }  

      String nome = MyIO.readLine();
      //Repeticao para fazer pesquisa de personagens
      while(!nome.equals("FIM")) {
         if(p.pesquisa(nome)){
            MyIO.println("SIM");
         }else{
            MyIO.println("NAO");
         }
         nome = MyIO.readLine();
      }  
      long fim = new Date().getTime();
      gravar.println("650525\t"+p.getNumComparacoes()+"\t"+(fim-inicio)/1000.0);//gravar no arquivo numero de
                                                                                //comparacoes e tempo de execucao
      arquivo.close();
   }
}

File: TP2/TP2Q08/StarWarsPesquisa.c
/*
Nome: Larissa Domingues Gomes
Turno: Manha
Numero de Matricula: 650525
Professor: Felipe Cunha 
Questao:Classe Personagens Star Wars C Pesquisa binaria
*/

#include "ListaPersonagem.h"
#include <time.h>

bool IsFim(char p[])
{
   return (strlen(p) == 3 && p[0] == 'F' && p[1] == 'I' && p[2] == 'M');
}
 
int main()
{
   clock_t t1 = clock();
   int comp = 0;
   FILE* arq = fopen("matrícula_binaria.txt", "wt");
   char leitura[_MAX_];
   ListaPersonagem l;
   ConstrutorLista(&l, 90);
   scanf("%s", leitura);
   getchar();//limpar buffer

   //Repeticao para fazer o set de dados do array de personagens ate o fim
   while(!IsFim(leitura)) {
      inserirFim(leitura, &l);
      scanf("%s", leitura);
      getchar();//limpar buffer
   }           
   scanf(" %[^\n]s", leitura);
  // getchar();
  //printf("%s\n",leitura);
   
   //Repeticao para procurar elementos na lista
   while(!IsFim(leitura)) {
      if(procurar(&l, leitura,&comp))
         printf("SIM\n");
      else
         printf("NAO\n");
      scanf(" %[^\n]s", leitura);
      //printf("%s\n",leitura);
   }                             
   freeLista(&l);
   clock_t t2 = clock();
   fprintf(arq, "650525\t%lf\t%i", (t2-t1)/1000.0, comp);
   fclose(arq);
}
